### PR TITLE
Return custom prefix for uri generation

### PR DESCRIPTION
**Pull request by bfidlers**

### CODE REVIEW

File: queries/formInstances.ts [hunk 1]
```diff
@@ -31,6 +31,36 @@ export const getFormLabel = async function (formTtl: string) {
...
+  const defaultPrefix = 'http://data.lblod.info/form-data/instances';
+
+  const bindings = await bindingStream.toArray();
+  if (bindings.length) {
+    const binding = bindings[0].get('o');
+    if (binding) {
+      return binding.value;
+    } else {
+      return defaultPrefix;
+    }
```
Comment (ruben-bottu) [inline queries/formInstances.ts:58]: Hopefully the 'o' will never be empty, but you never know I guess. the if can be shortened using the ternary operator though!

**Review by ruben-bottu — APPROVED** [review]

File: queries/formInstances.ts (hunk 1 above)
Comment (Rahien) [reply queries/formInstances.ts:58]: o can indeed still be the empty string, would be nice to require at least length 1 or something here

File: queries/formInstances.ts [hunk 2]
```diff
@@ -31,6 +31,36 @@ export const getFormLabel = async function (formTtl: string) {
...
+  const store = await ttlToStore(formTtl);
+  const engine = new QueryEngine();
+  const bindingStream = await engine.queryBindings(q, {
+    sources: [store],
+  });
+
+  const defaultPrefix = 'http://data.lblod.info/form-data/instances';
+
+  const bindings = await bindingStream.toArray();
+  if (bindings.length) {
```
Comment (Rahien) [inline queries/formInstances.ts:52]: nitpick, but thought i'd share as it's a nice pattern in general. I'm a big fan of the 'return early' pattern: https://medium.com/swlh/return-early-pattern-3d18a41bba8

This keeps your code on the 'happy path', minimizing the number of constraints you have to keep in mind while reading. i.e. in this case you would go:
```
if (bindings.length === 0) {
  return defaultPrefix;
}
// now we know we're on the happy path again!
const binding = bindings[0].get('o');
if(!binding || binding.length < 1) {
  throw new Error('someone did something really bad here, better fail hard');
}
// yay happy path again
return binding.value;
```

**Review by Rahien — CHANGES_REQUESTED** [review]
see the comment about the return early pattern

**Review by Rahien — APPROVED** [review]

**Pull request merged**